wit/request-reply.wit

```diff
@@ -0,0 +1,27 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
+    use types.{client, message, error};
+
+    /// Options for a request/reply operation
+    record request-options {
+        /// The maximum amount of time to wait for a response. If the timeout value is not set, then
+        /// the request/reply operation will block until a message is received in response.
+        timeout-ms: option<u32>,
+        /// The number of expected replies. If the number of replies is not set, then the
+        /// request/reply operation will block until a single message is received in response (or a
+        /// timeout value is reached). This can be used in conjunction for "scatter/gather" type
+        /// operations.
+        expected-replies: option<u32>,
+    }
+
+    /// Performs a blocking request/reply operation with an optional set of request options
+    request: func(c: client, msg: message, opts: option<request-options>) -> result<option<list<message>>, error>;
```

Comment:
I think you need `c` to be a `borrow<client>`.  Also, it seems like `expected-replies` is a pretty essential part of how this function works (it explains the `list` in the return type) so maybe it should be a proper parameter instead of being "hidden" in `request-options`?  It sounds like maybe it's not common to use a value other than 1, so what about improving the ergonomics of the common case with:
```
request: func(c: borrow<client>, m: message, opts: option<request-options>) -> result<mesage, error>;
request-list: func(c: borrow<client>, m: message, expected-replies: u32, opts: option<request-options>) -> result<list<message>, error>;
```